src/brushes/pencil_brush.class.js

```diff
@@ -335,9 +335,25 @@
       } while (checkIndex < fullPath.length - 1)
       // always push the line ending
       newPath.push(fullPath[fullPath.length - 1])
-      return newPath
+      return newPath;
+    },
+
+    detectAndEnlargeSinglePointPath: function (path){
+      var lastIndex = path.length - 1;
+      for (var i = 0; i < lastIndex; i++){
+        if (path[i][1] !== path[i + 1][1] ||
+            path[i][2] !== path[i + 1][2]) {
+          // path contains at least one divergent point
+          return path;
+        }
+      }
+      // we have detected only a single point. Add a tiny change to it so it will be rendered.
+      // Safari specs: https://html.spec.whatwg.org/multipage/canvas.html#trace-a-path 
+      // Specifically: 'Prune all zero-length line segments from path.'
+      var lastPoint = ['L', path[lastIndex][1], (path[lastIndex][2] + 1)];
```

Comment:
Just want to confirm that +1 here is enough. Is there any chance the +1, due to current scale, might end up being the same? Not sure if this is a valid concern as it depends how this +1 makes its way to the final API call.

Reply:
Yes, +1 is enough. (even +.001 is enough for the API ignoring our own precision/rounding) - I double checked this and the path-coords themselves are not modified by the scale. The calls to 'ctx.lineTo(x,y)' are the same regardless of screen size. I also just checked retina vs non and verified that the whole-number is unaffected.